AirCasting/Graph/AirCastingGraph.swift

```diff
@@ -179,7 +179,9 @@ extension AirCastingGraph {
         noteButtons.forEach { $0.button.removeFromSuperview() }
         noteButtons = notes.map { (createNewNoteButton(), $0, onTap) }
         noteButtons.forEach { lineChartView.addSubview($0.button) }
-        setNeedsLayout()
+        DispatchQueue.main.async { [self] in
```

Comment:
if this `setupNotes` func is not called on main thread other lines here are also manipulating the UI, so it should be wrapped in the dispatch block too. But I don't think that's a good idea, the caller should be responsible for making sure it's main thread. Let's add an assert at the top and adjust the calling code:
```Swift
assert(Thread.isMainThread) // don't remember exact naming here, but something like that.
```